update_lunch.py

```diff
@@ -0,0 +1,123 @@
+    in_event = False # Whether the current line is in an event
+    properties = {} # When properties are discovered, they will be stuffed in here
+    events = [] # The list of all properties objects
+    last_prop_name = None
+    for line in lines:
+        if line == "BEGIN:VEVENT":
+            in_event = True
+        elif line == "END:VEVENT":
+            in_event = False
+            events.append(properties)
+            properties = {}
+        elif in_event:
+            print line
+            if line[0] == " ": # If the current line is a continuation of the previous line
+                properties[last_prop_name] += line[1:]
+            else: # If it is the start of a normal line
+                # Sample line: DTSTART;TZID=America/Los_Angeles:20151030T110500
+                colon_seperated_values = string.split(line, ":")
+
+                # Garbage anything between ; and :
+                last_prop_name = string.split(colon_seperated_values[0], ";")[0]
+
+                properties[last_prop_name] = colon_seperated_values[1]
+    return events
+
+def writeToDB(events): # Takes the raw events, sanitizes them, and plops them into the db
+    for event in events:
+        # Prettify the summary, date, and description
+
+        # Convert the datetime string (e.g. 20151124T233401) to a date object
+        format = "%Y%m%dT%H%M%S"
+        date = datetime.datetime.strptime(event["DTSTART"], format).date()
+
+        lunches_for_date = Lunch.query(Lunch.day == date)
+        logging.info(lunches_for_date)
+        has_lunch_for_date = False
+
+        for lunch in lunches_for_date: # If any results were returned
+            has_lunch_for_date = True
+
+        if (has_lunch_for_date):
+            logging.info(str(date) + " is already in the DB")
+            continue
+
+        # Remove the price and back slashes from the summary
+        summary = string.split(event["SUMMARY"], " | ")[0] # Remove the price
+        summary = summary.replace("\\", "") # Remove back slashes
+
+        # Remove html from the description, and break it up into lines
+        desc = event["DESCRIPTION"]
+        desc = desc.replace("\,", ",")
+        desc = desc.replace("\;", ";")
+        no_html_desc = re.sub("<.*?>", '', desc)
+        description = string.split(no_html_desc, '\\n')
+
+        entry = Lunch( \
+            summary=summary, \
+            description=description, \
+            day=date)
+        logging.info(str(entry))
+        entry.put()
+
+def updateDB():
+    url = "http://www.eastsideprep.org/?plugin=all-in-one-event-calendar&controller=ai1ec_exporter_controller&action=export_events&ai1ec_cat_ids=57?"
+    mainresponse = urllib2.urlopen(url)
+    #mainresponse = open("download.ics") # Opens downloaded ics file
+
+    text = mainresponse.read()
+    lines = text.splitlines()
+    events = getEvents(lines)
+    writeToDB(events)
+
+def getLunchForDate(days_into_past=7): # date is a date object
```

Comment:
This name isn't quite right. You get multiple lunches, and you're not passing in a date, but a range.

Maybe query_lunches? or read_lunches?